lib/sms/string_conversion.c

```diff
@@ -0,0 +1,437 @@
+ */
+uint8_t string_conversion_ascii_to_gsm7bit(
+	const uint8_t *p_data,
+	uint8_t  data_len,
+	uint8_t *p_out_data,
+	uint8_t *p_out_bytes,
+	uint8_t *p_out_chars,
+	bool     packing)
+{
+	uint8_t i_ascii = 0;
+	uint8_t i_7bit = 0;
+	uint8_t out_bytes = 0;
+	uint8_t char_7bit;
+	uint8_t char_ascii;
+
+	if ((p_data != NULL) && (p_out_data != NULL)) {
+		for (i_ascii = 0; i_ascii < data_len; i_ascii++) {
+			char_ascii = p_data[i_ascii];
+			char_7bit = ascii_to_7bit_table[char_ascii];
+
+			if ((char_7bit & STR_7BIT_ESCAPE_IND) == 0) {
+				/* Character is in default alphabet table */
+				if (i_7bit < STR_MAX_CHARACTERS) {
+					p_out_data[i_7bit++] = char_7bit;
+				} else {
+					break;
+				}
+			} else {
+				/* Character is in default extension table */
+				if (i_7bit < STR_MAX_CHARACTERS - 1) {
+					p_out_data[i_7bit++] = STR_7BIT_ESCAPE_CODE;
+					p_out_data[i_7bit++] = (char_7bit & STR_7BIT_CODE_MASK);
+				} else {
+					break;
+				}
+			}
+		}
+
+		if (packing == true) {
+			out_bytes = string_conversion_7bit_sms_packing(p_out_data, i_7bit);
+		} else {
+			out_bytes = i_7bit;
+		}
+	}
+
+	if (p_out_bytes != NULL) {
+		*p_out_bytes = out_bytes;
+	}
+	if (p_out_chars != NULL) {
+		*p_out_chars = i_7bit;
+	}
+	return i_ascii;
+}
+
+/**
+ * @brief Convert GSM 7 bit Default Alphabet characters to ASCII characters.
+ *
+ * @details gsm7bit_to_ascii_table conversion table is used. Perform also unpacking of
+ * the 7 bit string before conversion, if caller indicates that the string is packed.
+ *
+ * References: 3GPP TS 23.038 chapter 6.2.1: GSM 7 bit Default Alphabet
+ *
+ * @param[in] p_data Pointer to array of characters to be converted. No null termination.
+ * @param[out] p_out_data Pointer to buffer for the converted string. Shall have allocation
+ *             of at least "num_char" bytes. Note that this function does not add
+ *             null termination at the end of the string. It should be done by caller,
+ *             when needed. (In that case it could be useful to actually allocate
+ *             num_char+1 bytes here.)
+ * @param[in] num_char Number of 7-bit characters to be unpacked, including possible escape codes.
+ *            Also indicates maximum allowed number of characters to be stored to output
+ *            buffer by this function.
+ * @param[in] packed True if the 7bit string is packed, i.e. has to be unpacked before conversion.
+ *
+ * @return Number of valid bytes/characters in "p_out_data". May be less than "num_char" in the
+ *         case that the 7 bit string contains "escape/extended code" sequences, that are converted
+ *         to single ASCII characters.
+ *
+ *********************************************************************************************/
+uint8_t string_conversion_gsm7bit_to_ascii(
+	const uint8_t *p_data,
+	uint8_t *p_out_data,
+	uint8_t  num_char,
+	bool     packed)
+{
+	uint8_t *p_7bit;
+	uint8_t i_ascii = 0;
+	uint8_t i_7bit = 0;
```

Comment:
Remove prefixes indicating the type of the variable

Reply:
Done